sphinx/themes/basic/static/searchtools.js

```diff
@@ -496,9 +496,13 @@ const Search = {
 
       // create the mapping
       files.forEach((file) => {
-        if (fileMap.has(file) && fileMap.get(file).indexOf(word) === -1)
-          fileMap.get(file).push(word);
-        else fileMap.set(file, [word]);
+        if (fileMap.has(file)) {
+          if (fileMap.get(file).indexOf(word) === -1) {
+            fileMap.get(file).push(word);
+          }
+        } else {
+          fileMap.set(file, [word]);
+        }
```

Comment:
Please check my logic on this; would the following be equivalent?

```
if (!fileMap.has(file)) {
  fileMap.set(file, []);
}
if (fileMap.get(file).indexOf(word) === -1) {
  fileMap.get(file).push(word);
}
```

I think I translate the logic as: ensure that each `file, word` combination encountered is created uniquely in the filemap.

Reply:
Yes, I think that should work and would be less confusing!

Reply:
Actually, there's one minor bug in the above, should be this:

```suggestion
        if (!fileMap.has(file)) {
          fileMap.set(file, [word]);
        } else if (fileMap.get(file).indexOf(word) === -1) {
            fileMap.get(file).push(word);
        }
```

Reply:
Yep, that seems good to me.  One less comparison in the not-yet-existing-file case, and still fairly straightforward to reason about :+1: